mesonbuild/backend/vs2010backend.py

```diff
@@ -280,6 +280,16 @@ def get_target_deps(self, t, recursive=False):
                         all_deps[ldep.get_id()] = ldep.target
                     else:
                         all_deps[ldep.get_id()] = ldep
+
+                for ldep in target.link_depends:
+                    if isinstance(ldep, build.CustomTargetIndex):
+                        all_deps[ldep.get_id()] = ldep.target
+                    elif isinstance(ldep, File):
+                        assert not ldep.is_built
+                        pass
```

Comment:
You don't need `pass` here, pass is only necessary for cases where a body would be empty otherwise.

But more to the point, why are we asserting not built here?

Reply:
I unfortunately haven't gotten the test pass on windows (nor all on linux, w/ debian unstable), even without any modification. Not quite sure what the issue is. Hence doing part of the debugging via CI :(.

> Also, why are we asserting not built here?

Well, with msbuild we can't (or at least we don't know how yet, it's probably possible *somehow*) have dependencies on files, we need the targets.  And before the changes to build.py to process_link_depends() and the vs_module_defs handling link_depends always contained files, even when targets were passed, because that's sufficient for ninja. So I tried to change it so that only static files are in link_depends, not ones we might need a dependency for. Hence the assert.

But doing this in the vs backend is a bad idea, not sure what I was thinking, it should of course be somewhere in build.py.

That might have showed me more quickly that we can't make that assertion, because of configure_file(). As the test failing in `test cases/windows/9 vs module defs generated/subdir/meson.build` shows.

I don't yet know how dependencies on configure_file are handled.

Reply:
> I don't yet know how dependencies on configure_file are handled.

Since they presumably are static once meson generated, they shouldn't be needed to handle project dependencies I assume. But depending on what kind of target it's the input to, it might still be needed to trigger rebuilds. I guess msbuild's dependency tracking would do the trick for compile / link tasks, and custom targets shouldn't have link_depends. 

Brr, msbuild, brr.

Reply:
configure_file is processed exclusively during `meson setup` and the input file is added as a reconfigure dependency so edits to the source.in will trigger a full buildsystem reconfigure.

This is usually more or less okay, since the expected use of a configure_file is to configure that file with with a configuration_data() object which only exists in meson.

Is that what you mean?